src/styled/Common.ts

```diff
@@ -66,10 +40,115 @@ const Color = css<{
         | 'negative'
         | 'other-green'
         | 'other-red';
-    background: 'dark1' | 'dark2' | 'dark3' | 'dark4';
-}>`
+    background?: 'dark1' | 'dark2' | 'dark3' | 'dark4';
+}
+export const Color = css<ColorProps>`
     ${({ color }) => color && `color: var(--${color})`};
     ${({ background }) =>
         background && `background-color: var(--${background})`};
 `;
-export { FlexContainer, GridContainer, Font, FontSize, Color };
+
+interface PaddingProps {
+    padding?: string;
+}
+export const Padding = css<PaddingProps>`
+    ${({ padding }) => padding && `padding: ${padding}`};
+`;
+
+interface MarginProps {
+    margin?: string;
+}
+export const Margin = css<MarginProps>`
+    ${({ margin }) => margin && `margin: ${margin}`};
+`;
+
+//   ------------------------------ DISPLAY ---------------------------------------
+
+// Define the prop types for the GridContainer
+interface GridProps {
+    numCols?: number;
+    numRows?: number;
+    gapSize?: number;
+    fullHeight?: boolean;
+    fullWidth?: boolean;
+}
+const Grid = css<GridProps>`
+    display: grid;
+    grid-template-columns: ${({ numCols }) =>
+        numCols ? `repeat(${numCols}, 1fr)` : 'auto'};
+    grid-template-rows: ${({ numRows }) =>
+        numRows ? `repeat(${numRows}, 1fr)` : 'auto'};
+    gap: ${({ gapSize }) => (gapSize ? `${gapSize}px` : '0')};
+    ${({ fullHeight }) => (fullHeight ? 'height: 100%;' : '')}
+    ${({ fullWidth }) => (fullWidth ? 'width: 100%;' : '')}
+`;
+export const GridContainer = styled.div<
+    GridProps &
+        FontProps &
+        FontSizeProps &
+        FontWeightProps &
+        ColorProps &
+        PaddingProps &
+        MarginProps
+>`
+    ${Grid}
+    ${Font}
+    ${FontSize}
+    ${FontWeight}
+    ${Color}
+    ${Padding}
+    ${Margin}
+`;
+
+// Define the prop types for the FlexContainer
+interface FlexProps {
+    gap?: number;
+    fullHeight?: boolean;
+    fullWidth?: boolean;
+    flexDirection?: 'row' | 'column';
+    justifyContent?: string;
+    alignItems?: string;
+    overflow?: string;
+}
+const Flex = css<FlexProps>`
```

Comment:
What do you think of rewriting this likeso for readability?

```
const Flex = css<FlexProps>`
  ${({ gap, fullWidth, fullHeight, flexDirection, justifyContent, alignItems, overflow }) => {
    return `
      display: flex;
     ${gap && `gap: ${gap}px;`}
     ${fullWidth && 'width: 100%;'}
     ...
    `;
  }}
`;
```

Reply:
I noticed I was struggling to find what would be the defaults at a glance just bc of the noise